master/internal/api_workspace.go

```diff
@@ -600,17 +790,43 @@ func (a *apiServer) DeleteWorkspace(
 	if err != nil {
 		return nil, fmt.Errorf("error deleting workspace (%d) templates: %w", req.Id, err)
 	}
+	// Delete workspace-namespace binding.
+	_, err = db.Bun().NewDelete().
+		Model(&model.WorkspaceNamespace{}).
+		Where("workspace_id =  ?", req.Id).
+		Exec(ctx)
+
+	// Delete the workspace in Kubernetes.
+	var wsToDelete model.Workspace
+	err = db.Bun().NewSelect().Model(&model.Workspace{}).Where("id = ?", req.Id).Scan(ctx, &wsToDelete)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
nit: create a `DeleteWorkspace` function in postgres_workspace.go -- better to maintain all DB reads/writes & keep tests there. A SQL bug here would be hard to debug